Plugins/FpeMonitoring/src/FpeMonitor.cpp

```diff
@@ -0,0 +1,344 @@
+  std::vector<FpeInfo> copy{};
+  copy = std::move(m_stracktraces);
+  m_stracktraces.clear();
+
+  for (auto &info : copy) {
+    auto it = std::find_if(m_stracktraces.begin(), m_stracktraces.end(),
+                           [&info](const FpeInfo &el) {
+                             const auto &fl = *el.st->begin();
+                             const auto &fr = *info.st->begin();
+                             return el.type == info.type &&
+                                    (boost::stacktrace::hash_value(fl) ==
+                                     boost::stacktrace::hash_value(fr));
+                           });
+    if (it != m_stracktraces.end()) {
+      it->count += info.count;
+      continue;
+    }
+    m_stracktraces.push_back({info.count, info.type, std::move(info.st)});
+  }
+}
+
+FpeMonitor::FpeMonitor()
+    : m_excepts{FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW | FE_UNDERFLOW} {
+  enable();
+}
+
+FpeMonitor::FpeMonitor(int excepts) : m_excepts(excepts) {
+  enable();
+}
+
+FpeMonitor::~FpeMonitor() {
+  disable();
+}
+
+void FpeMonitor::signalHandler(int /*signal*/, siginfo_t *si, void *ctx) {
+  if (stack().empty()) {
+    return;
+  }
+
+  FpeMonitor &fpe = *stack().top();
+  fpe.m_result.m_counts.at(si->si_code)++;
+
+  try {
+    // collect stack trace skipping 2 frames, which should be the signal handler
+    // and the calling facility. This might be platform specific, not sure
+    auto [buffer, remaining] = fpe.m_buffer.next();
+    std::size_t depth = boost::stacktrace::safe_dump_to(2, buffer, remaining);
+    std::size_t stored =
+        depth * sizeof(boost::stacktrace::frame::native_frame_ptr_t);
+    fpe.m_buffer.pushOffset(stored);  // record how much storage was consumed
+    fpe.m_recorded.emplace_back(
+        static_cast<FpeType>(si->si_code), buffer,
+        remaining);  // record buffer offset and fpe type
+
+  } catch (const std::bad_alloc &e) {
+    std::cout << "Unable to collect stack trace due to memory limit"
+              << std::endl;
+  }
+
+#if defined(__linux__) && defined(__x86_64__)
+  __uint16_t *cw = &((ucontext_t *)ctx)->uc_mcontext.fpregs->cwd;
+  *cw |= FPU_EXCEPTION_MASK;
+
+  __uint16_t *sw = &((ucontext_t *)ctx)->uc_mcontext.fpregs->swd;
+  *sw &= ~FPU_STATUS_FLAGS;
+
+  __uint32_t *mxcsr = &((ucontext_t *)ctx)->uc_mcontext.fpregs->mxcsr;
+  // *mxcsr |= SSE_EXCEPTION_MASK;  // disable all SSE exceptions
+  *mxcsr |= ((*mxcsr & SSE_STATUS_FLAGS) << 7);
+  *mxcsr &= ~SSE_STATUS_FLAGS;  // clear all pending SSE exceptions
+#else
+  (void)ctx;
+#endif
+}
+
+void FpeMonitor::enable() {
+#if defined(__linux__) && defined(__x86_64__)
+  ensureSignalHandlerInstalled();
+
+  // clear pending exceptions so they don't immediately fire
+  std::feclearexcept(m_excepts);
+
+  if (!stack().empty()) {
+    // unset previous except state
+    fedisableexcept(stack().top()->m_excepts);
+  }
+  // apply this stack
+  feenableexcept(m_excepts);
+
+  stack().push(this);
+#else
+  (void)m_excepts;
```

Comment:
will it complain that the member var is not user?
```suggestion
#else
  (void)m_excepts;
```

Reply:
I got a warning, yes. 
I don't see what the suggested change is?